src/Compilers/Core/Portable/Generated/Operations.xml.Generated.cs

```diff
@@ -6548,28 +6548,59 @@ internal sealed partial class LazyTranslatedQueryExpression : BaseTranslatedQuer
         protected override IOperation ExpressionImpl => _lazyExpression.Value;
     }
 
-    internal sealed partial class FlowCapture : Operation, IFlowCaptureOperation
+    internal sealed partial class FlowCaptureReference : Operation, IFlowCaptureReferenceOperation
     {
-        public FlowCapture(int id, SyntaxNode syntax, bool isInitialization, ITypeSymbol type, Optional<object> constantValue) :
-            base(OperationKind.FlowCapture, semanticModel: null, syntax, type, constantValue, isImplicit: true)
+        public FlowCaptureReference(int id, SyntaxNode syntax, ITypeSymbol type, Optional<object> constantValue) :
+            base(OperationKind.FlowCaptureReference, semanticModel: null, syntax, type, constantValue, isImplicit: true)
         {
             Id = id;
-            IsInitialization = isInitialization;
         }
 
         public int Id { get; }
-        public bool IsInitialization { get; }
         public override IEnumerable<IOperation> Children
         {
             get
             {
                 return Array.Empty<IOperation>();
             }
         }
+
+        public override void Accept(OperationVisitor visitor)
+        {
+            visitor.VisitFlowCaptureReference(this);
+        }
+
+        public override TResult Accept<TArgument, TResult>(OperationVisitor<TArgument, TResult> visitor, TArgument argument)
+        {
+            return visitor.VisitFlowCaptureReference(this, argument);
+        }
+    }
+
+    internal sealed partial class FlowCapture : Operation, IFlowCaptureOperation
+    {
+        public FlowCapture(int id, SyntaxNode syntax, IOperation value) :
+            base(OperationKind.FlowCapture, semanticModel: null, syntax, type: null, constantValue: default, isImplicit: true)
```

Comment:
>type: null [](start = 73, length = 10)

I'm torn on whether I agree with making type always null for FlowCaptures. Currently, I'm thinking they should match the underlying type of the IOperation, especially since FlowCaptureReference does. What was your reasoning for having it be null?

Reply:
> I'm torn on whether I agree with making type always null for FlowCaptures. Currently, I'm thinking they should match the underlying type of the IOperation, especially since FlowCaptureReference does. What was your reasoning for having it be null?

These nodes do not represent values and, therefore, don't have types. 

---
In reply to: [163335769](https://github.com/dotnet/roslyn/pull/24358#discussion_r163335769) [](ancestors = 163335769)